pkg/operator/controllers/allocation_utils.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"net/http"
+	"regexp"
+
+	"github.com/go-logr/logr"
+)
+
+// AllocateArgs contains information necessary to allocate a GameServer
+type AllocateArgs struct {
+	SessionID      string   `json:"sessionID"`
+	BuildID        string   `json:"buildID"`
+	SessionCookie  string   `json:"sessionCookie"`
+	InitialPlayers []string `json:"initialPlayers"`
+}
+
+// isValidUUID returns true if the string is a valid UUID
+func isValidUUID(uuid string) bool {
+	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+	return r.MatchString(uuid)
+}
+
+// validateAllocateArgs validates an instance of the AllocateArgs struct.
+func validateAllocateArgs(aa *AllocateArgs) bool {
+	if !isValidUUID(aa.SessionID) || !isValidUUID(aa.BuildID) {
+		return false
+	}
+	return true
+}
+
+// RequestMultiplayerServerResponse contains details that are returned on a successful GameServer allocation call
+type RequestMultiplayerServerResponse struct {
+	IPV4Address string
+	Ports       string
+	SessionID   string
+}
+
+// internalServerError is a helper function for returning an internal server error
+func internalServerError(w http.ResponseWriter, l logr.Logger, err error, msg string) {
+	l.Error(err, msg)
+	w.WriteHeader(http.StatusInternalServerError)
+	w.Write([]byte("500 - " + msg + " " + err.Error()))
```

Comment:
Did you see the "Cross-site scripting vulnerability" warning?

Reply:
yeah, I *think* that this was there already (medium severity) (https://github.com/PlayFab/thundernetes/security/code-scanning). I also *think* that since these calls are to be made by an authenticated client per cluster, this is of small severity. We can file an issue to fix it soon (and my bad for not having filed it already)